### PR TITLE
Upgrade to Polkadot v1.1.0

### PR DESCRIPTION
**Pull request by jasl**
> We need a better way to show upstream changes because 1.0.0 to 1.2.0 has 5000+ commits and combined Cumulus and Polkadot
> Commits in polkadot-sdk repo are mismatched with the original, the 1.0.0 branch starts from https://github.com/paritytech/substrate/commit/2b76b44e6f7e948c8dffb7ff72d349f0702f5acf but in polkadot-sdk, it's https://github.com/paritytech/polkadot-sdk/commit/d0215bb04f7ec6b56111dc4dda5020e351424332

- https://github.com/paritytech/polkadot-sdk/compare/d0215bb...release-polkadot-v1.2.0
- https://github.com/paritytech/polkadot-sdk/commits/release-polkadot-v1.2.0/substrate/frame/contracts

Breaking changes:
- https://github.com/paritytech/substrate/pull/14745
- https://github.com/paritytech/substrate/pull/14612

Fixes:
- https://github.com/paritytech/polkadot-sdk/pull/1687

Notable changes:
- https://github.com/paritytech/substrate/pull/14731
- https://github.com/paritytech/substrate/pull/14750
- https://github.com/paritytech/substrate/pull/14020
- https://github.com/paritytech/substrate/pull/14423

In addition:
- Upgrade Rust to 1.73.0

### CODE REVIEW

**Comment by jasl** [comment]
```
error[E0432]: unresolved import `sp_state_machine::StorageTransactionCache`
  --> crates/pink/runtime/src/storage/mod.rs:10:5
   |
10 |     StorageTransactionCache,
   |     ^^^^^^^^^^^^^^^^^^^^^^^ no `StorageTransactionCache` in the root

error[E0407]: method `deposit_address` is not a member of trait `AddressGenerator`
   --> crates/pink/runtime/src/runtime/pallet_pink.rs:122:9
    |
122 | /         fn deposit_address(contract_addr: &T::AccountId) -> T::AccountId {
123 | |             let entropy = (b"contract_depo_v1", contract_addr).using_encoded(T::Hashing::hash);
124 | |             UncheckedFrom::unchecked_from(entropy)
125 | |         }
    | |_________^ not a member of trait `AddressGenerator`

error[E0437]: type `Overlay` is not a member of trait `TrieBackendStorage`
  --> crates/pink/runtime/src/storage/external_backend.rs:19:5
   |
19 |     type Overlay = MemoryDB<Hashing>;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not a member of trait `TrieBackendStorage`

error[E0220]: associated type `Transaction` not found for `Self`
  --> crates/pink/runtime/src/storage/mod.rs:16:69
   |
16 |     fn commit_transaction(&mut self, root: Hash, transaction: Self::Transaction);
   |                                                                     ^^^^^^^^^^^ associated type `Transaction` not found

error[E0107]: missing generics for struct `OverlayedChanges`
  --> crates/pink/runtime/src/storage/mod.rs:37:31
   |
37 |     ) -> (R, ExecSideEffects, OverlayedChanges) {
   |                               ^^^^^^^^^^^^^^^^ expected 1 generic argument
   |
note: struct defined here, with 1 generic parameter: `H`
  --> /Users/jasl/.cargo/git/checkouts/polkadot-sdk-cff69157b985ed76/f60318f/substrate/primitives/state-machine/src/overlayed_changes/mod.rs:92:12
   |
92 | pub struct OverlayedChanges<H: Hasher> {
   |            ^^^^^^^^^^^^^^^^ -
help: add missing generic argument
   |
37 |     ) -> (R, ExecSideEffects, OverlayedChanges<H>) {
   |                                               +++

error[E0107]: missing generics for struct `OverlayedChanges`
  --> crates/pink/runtime/src/storage/mod.rs:69:48
   |
69 |     pub fn changes_transaction(&self, changes: OverlayedChanges) -> (Hash, Backend::Transaction) {
   |                                                ^^^^^^^^^^^^^^^^ expected 1 generic argument
   |
note: struct defined here, with 1 generic parameter: `H`
  --> /Users/jasl/.cargo/git/checkouts/polkadot-sdk-cff69157b985ed76/f60318f/substrate/primitives/state-machine/src/overlayed_changes/mod.rs:92:12
   |
92 | pub struct OverlayedChanges<H: Hasher> {
   |            ^^^^^^^^^^^^^^^^ -
help: add missing generic argument
   |
69 |     pub fn changes_transaction(&self, changes: OverlayedChanges<H>) -> (Hash, Backend::Transaction) {
   |                                                                +++

error[E0220]: associated type `Transaction` not found for `Backend`
  --> crates/pink/runtime/src/storage/mod.rs:69:85
   |
69 |     pub fn changes_transaction(&self, changes: OverlayedChanges) -> (Hash, Backend::Transaction) {
   |                                                                                     ^^^^^^^^^^^ associated type `Transaction` not found

error[E0107]: missing generics for struct `OverlayedChanges`
  --> crates/pink/runtime/src/storage/mod.rs:84:47
   |
84 |     pub fn commit_changes(&mut self, changes: OverlayedChanges) {
   |                                               ^^^^^^^^^^^^^^^^ expected 1 generic argument
   |
note: struct defined here, with 1 generic parameter: `H`
  --> /Users/jasl/.cargo/git/checkouts/polkadot-sdk-cff69157b985ed76/f60318f/substrate/primitives/state-machine/src/overlayed_changes/mod.rs:92:12
   |
92 | pub struct OverlayedChanges<H: Hasher> {
   |            ^^^^^^^^^^^^^^^^ -
help: add missing generic argument
   |
84 |     pub fn commit_changes(&mut self, changes: OverlayedChanges<H>) {
   |                                                               +++

error[E0220]: associated type `Transaction` not found for `Self`
  --> crates/pink/runtime/src/storage/external_backend.rs:27:73
   |
27 |     fn commit_transaction(&mut self, root: Hash, mut transaction: Self::Transaction) {
   |                                                                         ^^^^^^^^^^^ associated type `Transaction` not found

Some errors have detailed explanations: E0107, E0220, E0407, E0432, E0437.
wip
For more information about an error, try `rustc --explain E0107`.
error: could not compile `pink` (lib) due to 9 previous errors
warning: build failed, waiting for other jobs to finish...
error[E0308]: mismatched types
   --> crates/phactory/api/src/storage_sync.rs:258:43
    |
258 |         storage.apply_changes(state_root, transaction);
    |                 -------------             ^^^^^^^^^^^ expected `MemoryDB<BlakeTwo256, HashKey<...>, ...>`, found `MemoryDB<BlakeTwo256, PrefixedKey<...>, ...>`
    |                 |
    |                 arguments to this method are incorrect
    |
    = note: `MemoryDB<BlakeTwo256, PrefixedKey<...>, ...>` and `MemoryDB<BlakeTwo256, HashKey<...>, ...>` have similar names, but are actually distinct types
note: `MemoryDB<BlakeTwo256, PrefixedKey<...>, ...>` is defined in crate `memory_db`
   --> /Users/jasl/.cargo/registry/src/index.crates.io-6f17d22bba15001f/memory-db-0.32.0/src/lib.rs:83:1
    |
83  | pub struct MemoryDB<H, KF, T>
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: `MemoryDB<BlakeTwo256, HashKey<...>, ...>` is defined in crate `phala_trie_storage`
   --> /Users/jasl/Workspaces/BlockChain/phala-blockchain/crates/phala-trie-storage/src/memdb.rs:51:1
    |
51  | pub struct MemoryDB<H, KF, T>
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: method defined here
   --> /Users/jasl/Workspaces/BlockChain/phala-blockchain/crates/phala-trie-storage/src/lib.rs:155:12
    |
155 |     pub fn apply_changes(&mut self, root: H::Out, transaction: MemoryDB<H>) {
    |            ^^^^^^^^^^^^^
```

File: standalone/runtime/src/lib.rs [hunk 1]
```diff
@@ -634,7 +639,7 @@ parameter_types! {
 
     pub BetterUnsignedThreshold: Perbill = Perbill::from_rational(1u32, 10_000);
 
-    // worker configs
+    // miner configs
```
Comment (kvinwang) [inline standalone/runtime/src/lib.rs:642]: I remember we have a PR removing all the miner/mining keywords.

File: standalone/node/src/service.rs [hunk 2]
```diff
@@ -171,13 +175,13 @@ pub fn new_partial(
...
-			executor,
-		)?;
-	let client = Arc::new(client);
+    let (client, backend, keystore_container, task_manager) =
+        sc_service::new_full_parts::<Block, RuntimeApi, _>(
+            config,
+            telemetry.as_ref().map(|(_, telemetry)| telemetry.handle()),
+            executor,
+        )?;
+    let client = Arc::new(client);
```
Comment (kvinwang) [inline standalone/node/src/service.rs:184]: So, does the upstream switched to spaces from tabs for indentation?

File: standalone/runtime/src/lib.rs (hunk 1 above)
Comment (jasl) [reply standalone/runtime/src/lib.rs:642]: This from upstream https://github.com/paritytech/polkadot-sdk/blob/master/substrate/bin/node/runtime/src/lib.rs#L705

**Review by kvinwang — APPROVED** [review]

**Pull request merged**